src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostTestBase.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.AspNetCore.Razor.Test.Common.Workspaces;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Razor.Remote;
+using Microsoft.CodeAnalysis.Text;
+using Xunit.Abstractions;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor.Test.Cohost;
+
+public class CohostTestBase(ITestOutputHelper testOutputHelper) : WorkspaceTestBase(testOutputHelper)
+{
+    private readonly ITestOutputHelper _testOutputHelper = testOutputHelper;
+
+    private IRemoteServiceProvider? _remoteServiceProvider;
+
+    internal IRemoteServiceProvider RemoteServiceProvider => _remoteServiceProvider.AssumeNotNull();
+
+    protected override Task InitializeAsync()
+    {
+        _remoteServiceProvider = new ShortCircuitingRemoteServiceProvider(_testOutputHelper);
+
+        return base.InitializeAsync();
+    }
+
+    protected TextDocument CreateRazorDocument(string input)
+    {
+        var hostProject = TestProjectData.SomeProject;
+        var hostDocument = TestProjectData.SomeProjectComponentFile1;
+
+        var sourceText = SourceText.From(input);
+
+        var solution = Workspace.CurrentSolution.AddProject(ProjectInfo.Create(
+            ProjectId.CreateNewId(Path.GetFileNameWithoutExtension(hostProject.FilePath)),
```

Comment:
nit: Would you mind adding a `debugName: ` to the argument here? I had to double-check the signature of `ProjectId.CreateNewId(...)` to remind myself of what the path was being used for. Thanks in advance for helping me with my poor working memory. :smile: